isthmus/src/main/java/io/substrait/isthmus/SubstraitCalciteSchema.java

```diff
@@ -0,0 +1,284 @@
+package io.substrait.isthmus;
+
+import com.google.common.collect.ImmutableSet;
+import io.substrait.isthmus.SqlConverterBase.DefinedTable;
+import io.substrait.relation.NamedScan;
+import io.substrait.relation.Rel;
+import io.substrait.relation.RelCopyOnWriteVisitor;
+import io.substrait.type.NamedStruct;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.apache.calcite.jdbc.CalciteSchema;
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+import org.apache.calcite.schema.Schema;
+import org.apache.calcite.schema.Table;
+import org.apache.calcite.schema.impl.AbstractSchema;
+import org.apache.calcite.schema.lookup.LikePattern;
+import org.apache.calcite.schema.lookup.Lookup;
+import org.apache.calcite.schema.lookup.Named;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/**
+ * A subclass of the Calcite Schema for creation from a Substrait relation
+ *
+ * <p>Implementation note:
+ *
+ * <p>The external Isthmus API can take a function that will return the table schema when needed,
+ * rather than it being available up front.
+ *
+ * <p>This was implemented by a special subclass of the Calcite simple schena. Since this was
+ * changed in Caclite 1.39.0; it failed to work; the protected methods it extended from change.
```

Comment:
```suggestion
 * <p>This was implemented by a special subclass of the Calcite simple schema. Since this was
 * changed in Calcite 1.39.0; it failed to work; the protected methods it extended from changed.
```